src/toolkits/style_transfer/style_transfer.cpp

```diff
@@ -0,0 +1,354 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#include <toolkits/style_transfer/style_transfer.hpp>
+
+#include <map>
+#include <random>
+#include <string>
+
+#include <core/data/image/image_type.hpp>
+#include <model_server/lib/image_util.hpp>
+#include <model_server/lib/variant_deep_serialize.hpp>
+#include <toolkits/style_transfer/style_transfer_model_definition.hpp>
+
+namespace turi {
+namespace style_transfer {
+
+using turi::coreml::MLModelWrapper;
+using turi::neural_net::compute_context;
+using turi::neural_net::float_array_map;
+using turi::neural_net::float_scalar;
+using turi::neural_net::shared_float_array;
+
+namespace {
+
+constexpr size_t STYLE_TRANSFER_VERSION = 1;
+
+constexpr size_t DEFAULT_HEIGHT = 256;
+
+constexpr size_t DEFAULT_WIDTH = 256;
+
+constexpr size_t DEFAULT_BATCH_SIZE = 1;
+
+void populate_image(std::vector<float>::iterator& start_iter,
+                    std::vector<float>::iterator& end_iter,
+                    const unsigned char* src_ptr) {
+  while (start_iter != end_iter) {
+    *start_iter = *src_ptr / 255.f;
+    ++src_ptr;
+    ++start_iter;
+  }
+}
+
+void prepare_images(const image_type& image, std::vector<float>& array,
+                    size_t width, size_t height, size_t channels,
+                    size_t index) {
+  size_t image_size = height * width * channels;
+  size_t offset = index * image_size;
+
+  image_type resized_image =
+      image_util::resize_image(image, width, height, channels, true, 1);
+
+  const unsigned char* resized_image_ptr = resized_image.get_image_data();
+
+  std::vector<float>::iterator start_iter = array.begin() + offset;
+  std::vector<float>::iterator end_iter = start_iter + image_size;
+
+  populate_image(start_iter, end_iter, resized_image_ptr);
```

Comment:
FYI, instead of defining `populate_image`, this could also just be a call to `std::transform`:
```
std::transform(resized_image_ptr, resized_image_ptr + image_size, start_iter,
    [](unsigned char val) { return val / 255.f; });
```
https://en.cppreference.com/w/cpp/algorithm/transform

In this case, instead of passing in `start_iter` and `end_iter` (per my last comment), you could just pass in `start_iter`.

Reply:
WOAH, didn't know transform did that. That's cool!